### PR TITLE
Updates Text to use transient props

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?

Updates Text to use transient props

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

### CODE REVIEW

File: src/js/components/Text/Text.js [hunk 1]
```diff
@@ -70,10 +71,10 @@ const Text = forwardRef(
     const styledTextResult = (
       <StyledText
         as={!as && tag ? tag : as}
-        colorProp={color}
         aria-label={a11yTitle}
-        truncate={truncate}
-        size={size}
+        $colorProp={color}
```
Comment (jcfilben) [inline src/js/components/Text/Text.js:75]:
```suggestion
        $color={color}
```
Should we just rename this to `$color`?

File: src/js/components/Text/Text.js [hunk 2]
```diff
@@ -70,10 +71,10 @@ const Text = forwardRef(
...
+        $truncate={truncate}
+        $size={size}
```
Comment (jcfilben) [inline src/js/components/Text/Text.js:77]: Can we alphabetize these while we are here

File: src/js/components/Text/Text.js (hunk 1 above)
Comment (taysea) [reply src/js/components/Text/Text.js:75]: For some reason I was getting a type error on that one:
Screen Shot 2023-07-11 at 1 28 45 PM

Comment (taysea) [reply src/js/components/Text/Text.js:75]: Not sure if it's a special case?

File: src/js/components/Text/Text.js (hunk 2 above)
Comment (taysea) [reply src/js/components/Text/Text.js:77]: Yes, good call

Comment (taysea) [reply src/js/components/Text/Text.js]: Done.

File: src/js/components/Text/Text.js (hunk 1 above)
Comment (jcfilben) [reply src/js/components/Text/Text.js:75]: Hmm interesting, in that case I'm fine with leaving it as `colorProp` for now

**Comment by jcfilben** [comment]
Looks good just need to fix the merge conflict

**Review by jcfilben — APPROVED** [review]

**Pull request merged**